### PR TITLE
Add delegating methods for InteractEvent

### PR DESCRIPTION
**Pull request by CDFN**

### CODE REVIEW

**Comment by ImMorpheus** [comment]
See https://github.com/SpongePowered/SpongeAPI/issues/1975 for the discussion on this and a possible approach to the method doc

File: src/main/java/org/spongepowered/api/event/action/InteractEvent.java [hunk 1]
```diff
@@ -24,14 +24,69 @@
...
  * Base event for all interactions.
  */
 public interface InteractEvent extends Event {
-
+    /**
+     * Gets the fake {@link Player} who performed interaction.
+     *
+     * @return Fake player performing interaction
+     */
+    default Optional<Player> fakePlayer() {
```
Comment (kashike) [inline src/main/java/org/spongepowered/api/event/action/InteractEvent.java:47]: Is this really something that should be here? If so, this probably needs to either have (or point to) more documentation of what a fake player actually is, when one is available, etc.

**Review by kashike — CHANGES_REQUESTED** [review]

**Comment by Zidane** [comment]
@CDFN I am going to put this PR onto the back burner because you have, intentionally or not, shined a light onto a much bigger issue with our events.

The Cause system means our event metadata is completely dynamic...but the fact that it is cuts us both ways. Namely it makes it difficult for plugin devs to know what to reasonably expect from an event without simply listening to it and printing a cause. We are discussing as a team how best to tackle this issue without compromising the dynamic attributes that truly sets Sponge a part from the other platforms.

**Comment by ImMorpheus** [comment]
Hey! You seem to be running an unsupported version (API 8) of SpongeAPI. Support for API 8 ended in October 2023.

API-8 is now end-of-life. We will not be issuing any more API updates for version 8 of our API[^1]

As part of our legacy cleanup we are closing issues relating to API8/1.16
___

If you wish to move this forward, please rebase this PR on the current branch and reopen it.

[^1]: https://forums.spongepowered.org/t/sponge-status-update-17th-october-2023/42212#spongeapi-82-1

**Pull request closed without merging**